src/components/data-exploration/hierarchical-tree/HierarchicalTree.jsx

```diff
@@ -42,156 +43,31 @@ const HierarchicalTree = (props) => {
-            addToChildren(element.children, key);
-          }
-        });
-      };
-
-      addToChildren(newTreeData, dropKey);
-    }
+    // If rootNode, ignore
+    if (dragNode.rootNode) return;
 
-    // We dropped the item into a gap between two items.
-    if (info.dropToGap) {
-      // It can either be above or below an element.
-      const dropPosition = info.node.dragOverGapTop ? 'top' : 'bottom';
-
-      const addIntoGap = (haystack, needle) => transform(
-        haystack,
-        (result, objValue) => {
-          // We found the place to insert the dragged element.
-          if (objValue.key === needle) {
-            /**
-               * If a dragged object is a root node, it should only be
-               * droppable in between root nodes.
-               */
-            if (!objValue.rootNode && dragObj.rootNode === true) {
-              shouldUpdateState = false;
-              return false;
-            }
-
-            /**
-               * If a dragged object is not a root node, it should only be
-               * droppable in between non-root nodes.
-               */
-            if (!dragObj.rootNode && objValue.rootNode === true) {
-              shouldUpdateState = false;
-              return false;
-            }
-
-            if (objValue.children) {
-              // eslint-disable-next-line no-param-reassign
-              objValue.children = addIntoGap(objValue.children, needle);
-            }
-
-            /**
-               * Add the drag object into the tree according to where it was
-               * dropped.
-               */
-            if (dropPosition === 'top') {
-              result.push(dragObj);
-              result.push(objValue);
-            } else {
-              result.push(objValue);
-              result.push(dragObj);
-            }
-          } else {
-            /**
-               * We have not found the place to insert the element.
-               * Carry on building the new object until we are there.
-               */
-            if (objValue.children) {
-              // eslint-disable-next-line no-param-reassign
-              objValue.children = addIntoGap(objValue.children, needle);
-            }
-            result.push(objValue);
-          }
-
-          return true;
-        },
-        [],
-      );
+    // pos is a string e.g.: 0-0-1, each number is a position in a tree level
+    const posFromArray = dragNode.pos.split('-');
+    const posToArray = node.pos.split('-');
 
-      newTreeData = addIntoGap(newTreeData, dropKey);
-    }
+    const fromPosition = parseInt(posFromArray[2], 10);
 
-    if (shouldUpdateState) {
-      props.onHierarchyUpdate(newTreeData);
-    }
+    // If not in the same cellClass, ignore
+    if (!_.isEqual(posFromArray[1], posToArray[1])) return;
+
+    // If was dropped in same place, ignore
+    if (fromPosition === dropPosition) return;
+
+    // If not dropped in gap, ignore
+    if (!dropToGap && !node.rootNode) return;
```

Comment:
Why do you need the `!node.rootNode` condition?

Reply:
answered in https://github.com/hms-dbmi-cellenics/ui/pull/609#discussion_r791046456

Reply:
Can you add this information in a comment explaining why you need the `rootNode` check?